bgpd/bgpd.h

```diff
@@ -408,6 +409,16 @@ struct bgp {
 	/* Id space for automatic RD derivation for an EVI */
 	bitfield_t rd_idspace;
 
+	/* L3-VNI corresponding to this vrf */
+	vni_t l3vni;
+
+	/* router-mac to be used in mac-ip routes for this vrf */
+	struct ethaddr rmac;
+
+	/* vrf flags */
+	uint32_t vrf_flags;
+#define BGP_VRF_AUTO (1 << 0)
```

Comment:
maybe vrf_flags could have been  moved to a struct vrf * structure.

here this is just a remark. no need to move it now.

Reply:
I agree that we should have something simillar for lib/vrf as well. However, these particular flags are only specific to bgpd. bgpd has special handling for auto created VRF (VRF which is not yet created in the kernel).